### PR TITLE
Merge develop to Master

### PR DESCRIPTION
**Pull request by ieow**
Merge develop to Master
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

-
-
-

### CODE REVIEW

File: src/controllers/TorusController.ts [hunk 1]
```diff
@@ -321,8 +355,9 @@ export default class TorusController extends BaseController<TorusControllerConfi
       config: this.config.CurrencyControllerConfig,
       state: this.state.CurrencyControllerState,
     });
-    this.currencyController.updateConversionRate();
+    this.currencyController.updateQueryToken(Object.values(this.tokenInfoController.state.tokenInfoMap));
```
Comment (metallicalfa2) [inline src/controllers/TorusController.ts:358]: check

File: src/controllers/TorusController.ts [hunk 2]
```diff
@@ -413,7 +449,7 @@ export default class TorusController extends BaseController<TorusControllerConfi
 
     this.tokenInfoController.on("store", (state2) => {
       this.update({ TokenInfoState: state2 });
-      this.lastTokenRefresh = new Date(); // useful in UI
+      this.currencyController.updateQueryToken(Object.values(state2.tokenInfoMap), true);
```
Comment (metallicalfa2) [inline src/controllers/TorusController.ts:452]: check

File: src/controllers/TorusController.ts [hunk 3]
```diff
@@ -1354,6 +1538,33 @@ export default class TorusController extends BaseController<TorusControllerConfi
...
 
+  private async loginWithPrivateKey(req: Ihandler<LoginWithPrivateKeyParams>): Promise<{ success: boolean }> {
+    if (!req.params?.privateKey) throw new Error("Invalid Private Key");
+    this.keyringController.update({ wallets: [] });
+    this.preferencesController.update({ identities: {}, selectedAddress: "" });
+    const publicKey = await this.addAccount(req.params?.privateKey, req.params?.userInfo);
+    this.setSelectedAccount(publicKey);
+
+    if (!this.privateKey) throw new Error("Waller Error");
+    this.saveToOpenloginBackend({ privateKey: req.params?.privateKey, publicKey: this.selectedAddress });
```
Comment (ieow) [inline src/controllers/TorusController.ts:1549]: do we want to save to state-backend for login with private key ?

**Review by metallicalfa2 — APPROVED** [review]
lgtm

**Pull request merged**